src/Page/Form.purs

```diff
@@ -206,14 +211,19 @@ eval = case _ of
     H.modify_ _ { form = map { expr, evaled: _ } (fromEither evaled) }
 
   UpdateKey key val a -> do
-    H.modify_ \state -> state { values = Data.Map.insert key val state.values }
-    { form } <- H.get
+    { values, form } <- H.get
     case form of
-      Success { expr } ->
-        eval (EvalForm (Lynx.Data.Form.setValue key val expr) a)
-      _ -> pure a
+      Success { expr } -> do
+        case val of
+          Nothing -> do
+            H.modify_ _ { values = Data.Map.delete key values }
+            eval (EvalForm (Lynx.Data.Form.setValue key UserCleared expr) a)
+          Just v -> do
+            H.modify_ _ { values = Data.Map.insert key v values }
+            eval (EvalForm (Lynx.Data.Form.setValue key (UserInput $ Val v) expr) a)
+      _ -> pure a -- should this be possible?
```

Comment:
Well, it is possible with how our state is modeled. What I really meant is, should it be possible? Since if we don't have a form, how would we have any key/value to update here? But you also mentioned in another comment how after more recent changes, this query really serves no purpose, so I'll likely get rid of it and avoid the problem altogether

Reply:
Removing the `UpdateKey`/`UpdateValue` case won't remove your concern, it'll just push it into `EvalForm`. I think if we don't want to deal with `RemoteData _ _`, we'd have to take a fully realized `Page Expr` and have the parent component deal with `RemoteData _ _`. Both ways seem feasible to me (keeping it in this component vs. pushing to the parent), so I'm not sure which way we should push.